test/benchmarks/benchmark_crc32.cc

```diff
@@ -53,7 +53,7 @@ class crc32: public benchmark::Fixture {
         } \
         Bench(state, fptr); \
     } \
-    BENCHMARK_REGISTER_F(crc32, name)->Range(1, MAX_RANDOM_INTS_SIZE);
+    BENCHMARK_REGISTER_F(crc32, name)->Arg(1)->Arg(8)->Arg(12)->Arg(15)->Arg(16)->Arg(31)->Arg(32)->Arg(63)->Arg(64)->Arg(512)->Arg(4<<10)->Arg(32<<10)->Arg(256<<10)->Arg(4096<<10)
```

Comment:
Perhaps this should be a new instance of `BENCHMARK_REGISTER_F` with `name_small`

Reply:
Some of those sizes can go away again before merging. 15, 31 and 63 are there mostly to show how bad it really is right before the various cutoff points. But it might be a good idea to keep one that is unaligned as well. 12?

Reply:
@nmoinvaz If you would like to add a benchmark instance like that (with the above-mentioned changes), and use it for the crc32 and adler32, then that would be great.
Gives better coverage for the smaller sizes where dragons are most likely to be hiding, and drops some of the bigger sizes that are more unlikely to show much different behavior between sizes.
I am also not sure we need to go as far as 4MB, do we? I was thinking 1MB might be more than enough, unless we want to test memory bandwidth and such as well.

Reply:
Actually, I didn't see the `<<10`.  I think what you have is probably good. And can be used on adler32 as well. Perhaps you want to make it more readable? IMO I think leaving at 4MB is a good.